doc/designs/disable-stale-users.md

```diff
@@ -0,0 +1,147 @@
+https://pagure.io/freeipa/issue/5313 for reliability reasons.
+That attribute was not replicated either.
+Therefore distinguishing between stale and active accounts cannot be done with
+existing logon data.
+
+However, user passwords have to be changed regularly in these environments as
+well. A user who failed to change his password in time can be considered stale
+and therefore be disabled by the system.
+In this case, a grace period should be provided between password expiration and
+the actual account locking event.
+
+The DisableStaleUsers (DSU) tooling is provided to make sure no non-locked-out
+account has a password expired since N days, N being 7 by default.
+Installing the DSU tooling will not change the behavior of existing clusters.
+DSU has to be explicitely configured and enabled to start disabling stale
+accounts.
+
+Accounts are locked out by setting the nsAccountLock attribute
+(OID: 2.16.840.1.113730.3.1.610) to True.
+
+On very busy clusters, using a non-busy replica (like a hidden replica) for DSU
+is recommended.
+
+
+## Limitations
+
+If regulations call for stale/inactive accounts to be disabled within X
+number of days, then the password lifetime of the passwords for these
+accounts must be set to a value lower than X, minus the expected grace period
+mentioned above, minus the maximum amount of time between two invocations
+of the DSU tool.
+
+
+## Example
+
+Given the following requirements and facts:
+* stale accounts must be disabled after 90 days' inactivity
+* grace period is decided to be 8 days
+* the tool runs every day
+
+setting the maximum password lifetime to 80 days or less would meet the
+guidelines as 80+8+1 is less than 90.
+
+## Multiple runners
+
+DSU can be installed on more than a single replica. This provides redundancy.
+However, care must be taken that DSU is not triggered both:
+* at the same time
+* on the same accounts
+
+So a proper multi-DSU setup has to change both the time of the systemd timer
+the tool is run and the grace period.
+For instance:
+* having one DSU instance start working at 2AM and the other at 4AM
+* making sure the work of the first instance plus the associated replication
+  traffic completes in much less than 2 hours
+* having the first scheduled DSU instance configured to disable accounts with
+  a 7-day grace period and the second instance configured with a 8-day grace
+  period
+
+would make sure that the second DSU instances never disables any account
+unless the first one is offline.
+
+
+## How to Use DSU
+
+### Installation
+
+The Disable Stale Users tooling is contained in the freeipa-server-dsu
+package. 
+``dnf install freeipa-server-dsu`` will install that package.
+
+
+### Configuration
+
+The configuration file for DSU contains the following variables:
+
+```
+[dsu]
+active = False
+grace_period = 7d
+ignore_groups = admins
+log = True
+```
+
+Switch active to True for the DSU tool to disable (lock) accounts.
```

Comment:
active sounds more like "it's running". disable_accounts seems clearer to me.

Reply:
Would it be better to store the config in LDAP so it doesn't have to be updated on all masters running the service?

Reply:
Fixed, thanks for the proposal!